src/vcSettings.cpp

```diff
@@ -822,3 +827,24 @@ udResult vcSettings_UpdateLanguageOptions(vcSettings *pSetting)
 epilogue:
   return result;
 }
+
+void vcSettings_UpdateServerFlag(vcSettings *pSettings)
+{
+    UDCOMPILEASSERT(udLengthOf(g_vcMapTileServerName) == vcMTSF_Count, "Update array");
+
+    pSettings->maptiles.serverFlag = vcMTSF_Count;
+    for (size_t i = 0; i < vcMTSF_Count; i++)
+    {
+        if (udStrstr(pSettings->maptiles.tileServerAddress, udStrlen(pSettings->maptiles.tileServerAddress),g_vcMapTileServerName[i]) != 0)
```

Comment:
In this instance, since we know that tileServerAddress is null terminated, providing strlen is less efficient, preferably I'd pass a 0 in its place

Reply:
Ok, thanks for your comments.I'm not sure what you mean, but I've optimized it: reduce the number of strlen calls.